Source/WebCore/dom/ContextDestructionObserver.h

```diff
@@ -37,7 +37,7 @@ class ContextDestructionObserver {
 public:
     WEBCORE_EXPORT virtual void contextDestroyed();
 
-    ScriptExecutionContext* scriptExecutionContext() const { return m_scriptExecutionContext.get(); }
+    ScriptExecutionContext* scriptExecutionContext() const; // Defined in ContextDestructionObserverInlines.h.
```

Comment:
If we put the inline keyword here but no function body then we get a better error when we forget to include the inlines header.

Also this pattern is common enough that I’m not sure we should include the comment.

Reply:
Ah yes, I'll put it back. I had to remove it to sort out issues with the glib bots. While `inline` here generates nice errors on Mac bots, I found the errors from the glib bots very unhelpful.

Reply:
Odd, I added the `inline` again and now the glib bots are failing to build again with this error:
```
FAILED: Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/JSSpeechSynthesisErrorEvent.cpp.o 
/usr/bin/ccache /usr/bin/c++ -DBUILDING_GTK__=1 -DBUILDING_WEBKIT=1 -DBUILDING_WITH_CMAKE=1 -DBUILDING_WebCore -DBWRAP_EXECUTABLE=\"/usr/bin/bwrap\" -DDBUS_PROXY_EXECUTABLE=\"/usr/bin/xdg-dbus-proxy\" -DGETTEXT_PACKAGE=\"WebKitGTK-4.1\" -DHAVE_CONFIG_H=1 -DJSC_GLIB_API_ENABLED -DPAS_BMALLOC=1 -DSTATICALLY_LINKED_WITH_PAL -DUSE_SYSTEM_EGL -I/app/webkit/WebKitBuild/GTK/Release -I/app/webkit/WebKitBuild/GTK/Release/WebCore/DerivedSources -I/app/webkit/Source/WebCore -I/app/webkit/Source/WebCore/Modules/ShapeDetection -I/app/webkit/Source/WebCore/Modules/ShapeDetection/Interfaces -I/app/webkit/Source/WebCore/Modules/WebGPU -I/app/webkit/Source/WebCore/Modules/WebGPU/InternalAPI -I/app/webkit/Source/WebCore/Modules/WebGPU/Implementation -I/app/webkit/Source/WebCore/Modules/airplay -I/app/webkit/Source/WebCore/Modules/applepay -I/app/webkit/Source/WebCore/Modules/applepay/paymentrequest -I/app/webkit/Source/WebCore/Modules/applicationmanifest -I/app/webkit/Source/WebCore/Modules/async-clipboard -I/app/webkit/Source/WebCore/Modules/audiosession -I/app/webkit/Source/WebCore/Modules/badge -I/app/webkit/Source/WebCore/Modules/beacon -I/app/webkit/Source/WebCore/Modules/cache -I/app/webkit/Source/WebCore/Modules/compression -I/app/webkit/Source/WebCore/Modules/contact-picker -I/app/webkit/Source/WebCore/Modules/cookie-consent -I/app/webkit/Source/WebCore/Modules/cookie-store -I/app/webkit/Source/WebCore/Modules/credentialmanagement -I/app/webkit/Source/WebCore/Modules/encryptedmedia -I/app/webkit/Source/WebCore/Modules/encryptedmedia/legacy -I/app/webkit/Source/WebCore/Modules/entriesapi -I/app/webkit/Source/WebCore/Modules/fetch -I/app/webkit/Source/WebCore/Modules/filesystemaccess -I/app/webkit/Source/WebCore/Modules/geolocation -I/app/webkit/Source/WebCore/Modules/highlight -I/app/webkit/Source/WebCore/Modules/indexeddb -I/app/webkit/Source/WebCore/Modules/indexeddb/client -I/app/webkit/Source/WebCore/Modules/indexeddb/server -I/app/webkit/Source/WebCore/Modules/indexeddb/shared -I/app/webkit/Source/WebCore/Modules/mediacapabilities -I/app/webkit/Source/WebCore/Modules/mediacontrols -I/app/webkit/Source/WebCore/Modules/mediarecorder -I/app/webkit/Source/WebCore/Modules/mediasession -I/app/webkit/Source/WebCore/Modules/mediasource -I/app/webkit/Source/WebCore/Modules/mediastream -I/app/webkit/Source/WebCore/Modules/model-element -I/app/webkit/Source/WebCore/Modules/model-element/dummy -I/app/webkit/Source/WebCore/Modules/navigatorcontentutils -I/app/webkit/Source/WebCore/Modules/notifications -I/app/webkit/Source/WebCore/Modules/paymentrequest -I/app/webkit/Source/WebCore/Modules/permissions -I/app/webkit/Source/WebCore/Modules/pictureinpicture -I/app/webkit/Source/WebCore/Modules/plugins -I/app/webkit/Source/WebCore/Modules/push-api -I/app/webkit/Source/WebCore/Modules/remoteplayback -I/app/webkit/Source/WebCore/Modules/reporting -I/app/webkit/Source/WebCore/Modules/screen-wake-lock -I/app/webkit/Source/WebCore/Modules/speech -I/app/webkit/Source/WebCore/Modules/storage -I/app/webkit/Source/WebCore/Modules/streams -I/app/webkit/Source/WebCore/Modules/web-locks -I/app/webkit/Source/WebCore/Modules/webaudio -I/app/webkit/Source/WebCore/Modules/webauthn -I/app/webkit/Source/WebCore/Modules/webauthn/cbor -I/app/webkit/Source/WebCore/Modules/webauthn/fido -I/app/webkit/Source/WebCore/Modules/webcodecs -I/app/webkit/Source/WebCore/Modules/webdatabase -I/app/webkit/Source/WebCore/Modules/webdriver -I/app/webkit/Source/WebCore/Modules/websockets -I/app/webkit/Source/WebCore/Modules/webtransport -I/app/webkit/Source/WebCore/Modules/webxr -I/app/webkit/Source/WebCore/accessibility -I/app/webkit/Source/WebCore/accessibility/isolatedtree -I/app/webkit/Source/WebCore/animation -I/app/webkit/Source/WebCore/bindings -I/app/webkit/Source/WebCore/bindings/js -I/app/webkit/Source/WebCore/bridge -I/app/webkit/Source/WebCore/bridge/c -I/app/webkit/Source/WebCore/bridge/jsc -I/app/webkit/Source/WebCore/contentextensions -I/app/webkit/Source/WebCore/crypto -I/app/webkit/Source/WebCore/crypto/algorithms -I/app/webkit/Source/WebCore/crypto/keys -I/app/webkit/Source/WebCore/crypto/parameters -I/app/webkit/Source/WebCore/css -I/app/webkit/Source/WebCore/css/calc -I/app/webkit/Source/WebCore/css/color -I/app/webkit/Source/WebCore/css/parser -I/app/webkit/Source/WebCore/css/query -I/app/webkit/Source/WebCore/css/typedom -I/app/webkit/Source/WebCore/css/typedom/color -I/app/webkit/Source/WebCore/css/typedom/numeric -I/app/webkit/Source/WebCore/css/typedom/transform -I/app/webkit/Source/WebCore/cssjit -I/app/webkit/Source/WebCore/dom -I/app/webkit/Source/WebCore/dom/messageports -I/app/webkit/Source/WebCore/domjit -I/app/webkit/Source/WebCore/editing -I/app/webkit/Source/WebCore/fileapi -I/app/webkit/Source/WebCore/history -I/app/webkit/Source/WebCore/html -I/app/webkit/Source/WebCore/html/canvas -I/app/webkit/Source/WebCore/html/forms -I/app/webkit/Source/WebCore/html/parser -I/app/webkit/Source/WebCore/html/shadow -I/app/webkit/Source/WebCore/html/track -I/app/webkit/Source/WebCore/inspector -I/app/webkit/Source/WebCore/inspector/agents -I/app/webkit/Source/WebCore/inspector/agents/page -I/app/webkit/Source/WebCore/inspector/agents/worker -I/app/webkit/Source/WebCore/layout -I/app/webkit/Source/WebCore/layout/formattingContexts -I/app/webkit/Source/WebCore/layout/formattingContexts/block -I/app/webkit/Source/WebCore/layout/formattingContexts/block/tablewrapper -I/app/webkit/Source/WebCore/layout/formattingContexts/flex -I/app/webkit/Source/WebCore/layout/floats -I/app/webkit/Source/WebCore/layout/formattingContexts/inline -I/app/webkit/Source/WebCore/layout/formattingContexts/inline/display -I/app/webkit/Source/WebCore/layout/formattingContexts/inline/invalidation -I/app/webkit/Source/WebCore/layout/formattingContexts/inline/ruby -I/app/webkit/Source/WebCore/layout/formattingContexts/inline/text -I/app/webkit/Source/WebCore/layout/integration -I/app/webkit/Source/WebCore/layout/integration/flex -I/app/webkit/Source/WebCore/layout/integration/inline -I/app/webkit/Source/WebCore/layout/layouttree -I/app/webkit/Source/WebCore/layout/formattingContexts/table -I/app/webkit/Source/WebCore/loader -I/app/webkit/Source/WebCore/loader/appcache -I/app/webkit/Source/WebCore/loader/archive -I/app/webkit/Source/WebCore/loader/archive/mhtml -I/app/webkit/Source/WebCore/loader/cache -I/app/webkit/Source/WebCore/loader/icon -I/app/webkit/Source/WebCore/mathml -I/app/webkit/Source/WebCore/page -I/app/webkit/Source/WebCore/page/csp -I/app/webkit/Source/WebCore/page/scrolling -I/app/webkit/Source/WebCore/platform -I/app/webkit/Source/WebCore/platform/animation -I/app/webkit/Source/WebCore/platform/audio -I/app/webkit/Source/WebCore/platform/calc -I/app/webkit/Source/WebCore/platform/encryptedmedia -I/app/webkit/Source/WebCore/platform/gamepad -I/app/webkit/Source/WebCore/platform/graphics -I/app/webkit/Source/WebCore/platform/graphics/cpu/arm -I/app/webkit/Source/WebCore/platform/graphics/cpu/arm/filters -I/app/webkit/Source/WebCore/platform/graphics/displaylists -I/app/webkit/Source/WebCore/platform/graphics/filters -I/app/webkit/Source/WebCore/platform/graphics/filters/software -I/app/webkit/Source/WebCore/platform/graphics/controls -I/app/webkit/Source/WebCore/platform/graphics/iso -I/app/webkit/Source/WebCore/platform/graphics/opentype -I/app/webkit/Source/WebCore/platform/graphics/transforms -I/app/webkit/Source/WebCore/platform/mediacapabilities -I/app/webkit/Source/WebCore/platform/mediarecorder -I/app/webkit/Source/WebCore/platform/mediastream -I/app/webkit/Source/WebCore/platform/mock -I/app/webkit/Source/WebCore/platform/mock/mediasource -I/app/webkit/Source/WebCore/platform/network -I/app/webkit/Source/WebCore/platform/sql -I/app/webkit/Source/WebCore/platform/text -I/app/webkit/Source/WebCore/platform/xr -I/app/webkit/Source/WebCore/plugins -I/app/webkit/Source/WebCore/rendering -I/app/webkit/Source/WebCore/rendering/line -I/app/webkit/Source/WebCore/rendering/mathml -I/app/webkit/Source/WebCore/rendering/shapes -I/app/webkit/Source/WebCore/rendering/style -I/app/webkit/Source/WebCore/rendering/svg -I/app/webkit/Source/WebCore/rendering/svg/legacy -I/app/webkit/Source/WebCore/rendering/updating -I/app/webkit/Source/WebCore/replay -I/app/webkit/Source/WebCore/storage -I/app/webkit/Source/WebCore/style -I/app/webkit/Source/WebCore/svg -I/app/webkit/Source/WebCore/svg/animation -I/app/webkit/Source/WebCore/svg/graphics -I/app/webkit/Source/WebCore/svg/graphics/filters -I/app/webkit/Source/WebCore/svg/properties -I/app/webkit/Source/WebCore/websockets -I/app/webkit/Source/WebCore/workers -I/app/webkit/Source/WebCore/workers/service -I/app/webkit/Source/WebCore/workers/service/background-fetch -I/app/webkit/Source/WebCore/workers/service/context -I/app/webkit/Source/WebCore/workers/service/server -I/app/webkit/Source/WebCore/workers/shared -I/app/webkit/Source/WebCore/workers/shared/context -I/app/webkit/Source/WebCore/worklets -I/app/webkit/Source/WebCore/xml -I/app/webkit/Source/WebCore/xml/parser -I/app/webkit/Source/WebCore/Modules/gamepad -I/app/webkit/Source/WebCore/platform/encryptedmedia/clearkey -I/app/webkit/Source/WebCore/platform/graphics/angle -I/app/webkit/Source/WebCore/platform/graphics/lcms -I/app/webkit/Source/ThirdParty/xdgmime/src -I/app/webkit/Source/WebCore/platform/adwaita -I/app/webkit/Source/WebCore/platform/graphics/cairo -I/app/webkit/Source/WebCore/platform/graphics/freetype -I/app/webkit/Source/WebCore/platform/graphics/harfbuzz -I/app/webkit/Source/WebCore/platform/graphics/harfbuzz/ng -I/app/webkit/Source/WebCore/Modules/mediastream/gstreamer -I/app/webkit/Source/WebCore/platform/graphics/gstreamer -I/app/webkit/Source/WebCore/platform/graphics/gstreamer/mse -I/app/webkit/Source/WebCore/platform/graphics/gstreamer/eme -I/app/webkit/Source/WebCore/platform/gstreamer -I/app/webkit/Source/WebCore/platform/mediarecorder/gstreamer -I/app/webkit/Source/WebCore/platform/audio/gstreamer -I/app/webkit/Source/WebCore/platform/image-decoders -I/app/webkit/Source/WebCore/platform/image-decoders/avif -I/app/webkit/Source/WebCore/platform/image-decoders/bmp -I/app/webkit/Source/WebCore/platform/image-decoders/gif -I/app/webkit/Source/WebCore/platform/image-decoders/ico -I/app/webkit/Source/WebCore/platform/image-decoders/jpeg -I/app/webkit/Source/WebCore/platform/image-decoders/jpeg2000 -I/app/webkit/Source/WebCore/platform/image-decoders/jpegxl -I/app/webkit/Source/WebCore/platform/image-decoders/png -I/app/webkit/Source/WebCore/platform/image-decoders/webp -I/app/webkit/Source/WebCore/platform/network/soup -I/app/webkit/Source/WebCore/platform/graphics/texmap -I/app/webkit/Source/WebCore/platform/graphics/nicosia -I/app/webkit/Source/WebCore/page/scrolling/nicosia -I/app/webkit/Source/WebCore/platform/graphics/texmap/coordinated -I/app/webkit/Source/WebCore/platform/graphics/nicosia/cairo -I/app/webkit/Source/WebCore/platform/graphics/nicosia/texmap -I/app/webkit/Source/WebCore/accessibility/atspi -I/app/webkit/Source/WebCore/crypto/openssl -I/app/webkit/Source/WebCore/page/gtk -I/app/webkit/Source/WebCore/platform/audio/glib -I/app/webkit/Source/WebCore/platform/generic -I/app/webkit/Source/WebCore/platform/glib -I/app/webkit/Source/WebCore/platform/gtk -I/app/webkit/Source/WebCore/platform/graphics/egl -I/app/webkit/Source/WebCore/platform/graphics/epoxy -I/app/webkit/Source/WebCore/platform/graphics/gbm -I/app/webkit/Source/WebCore/platform/graphics/gtk -I/app/webkit/Source/WebCore/platform/graphics/opengl -I/app/webkit/Source/WebCore/platform/graphics/wayland -I/app/webkit/Source/WebCore/platform/graphics/x11 -I/app/webkit/Source/WebCore/platform/mediastream/gtk -I/app/webkit/Source/WebCore/platform/mediastream/gstreamer -I/app/webkit/Source/WebCore/platform/network/glib -I/app/webkit/Source/WebCore/platform/text/gtk -I/app/webkit/WebKitBuild/GTK/Release/ANGLE/Headers -I/app/webkit/WebKitBuild/GTK/Release/JavaScriptCore/Headers -I/app/webkit/WebKitBuild/GTK/Release/JavaScriptCore/PrivateHeaders -I/app/webkit/WebKitBuild/GTK/Release/JavaScriptCoreGLib/Headers -I/app/webkit/WebKitBuild/GTK/Release/JavaScriptCoreGLib/DerivedSources -I/app/webkit/WebKitBuild/GTK/Release/WTF/Headers -I/app/webkit/WebKitBuild/GTK/Release/bmalloc/Headers -I/app/webkit/WebKitBuild/GTK/Release/PAL/Headers -isystem /usr/include/gstreamer-1.0 -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem /usr/include/orc-0.4 -isystem /usr/lib/x86_64-linux-gnu/gstreamer-1.0/include -isystem /usr/include/blkid -isystem /usr/include/libmount -isystem /usr/include/WPEFramework/ocdm -isystem /usr/include/libsoup-3.0 -isystem /usr/include/enchant-2 -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/libsecret-1 -isystem /usr/include/flite -isystem /usr/include/libdrm -isystem /usr/include/libxml2 -isystem /usr/include/cairo -isystem /usr/include/freetype2 -isystem /usr/include/harfbuzz -isystem /usr/include/openjpeg-2.5 -isystem /usr/include/avif -isystem /usr/include/gtk-3.0 -isystem /usr/include/pango-1.0 -isystem /usr/include/gdk-pixbuf-2.0 -isystem /usr/include/dbus-1.0 -isystem /usr/lib/x86_64-linux-gnu/dbus-1.0/include -isystem /usr/include/atk-1.0 -isystem /usr/include/webp -isystem /usr/include/libpng16 -isystem /usr/include/pixman-1 -isystem /usr/include/fribidi -isystem /usr/include/at-spi2-atk/2.0 -isystem /usr/include/at-spi-2.0 -isystem /usr/include/libmanette -fdiagnostics-color=always -Wextra -Wall -pipe -fmax-errors=20 -Wno-odr -Wno-stringop-overread -Wno-stringop-overflow -Wno-nonnull -Wno-array-bounds -Wno-expansion-to-defined -Wno-noexcept-type -Wno-psabi -Wno-misleading-indentation -Wno-maybe-uninitialized -Wundef -Wpointer-arith -Wmissing-format-attribute -Wformat-security -Wcast-align -Wno-tautological-compare  -fno-strict-aliasing -fno-exceptions -fno-rtti -O3 -DNDEBUG -std=c++20 -fPIC -fvisibility=hidden -Werror -DAVIF_DLL -pthread -MD -MT Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/JSSpeechSynthesisErrorEvent.cpp.o -MF Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/JSSpeechSynthesisErrorEvent.cpp.o.d -o Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/JSSpeechSynthesisErrorEvent.cpp.o -c /app/webkit/WebKitBuild/GTK/Release/WebCore/DerivedSources/JSSpeechSynthesisErrorEvent.cpp
In file included from /app/webkit/Source/WebCore/dom/ActiveDOMObject.h:29,
                 from /app/webkit/Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h:30,
                 from /app/webkit/Source/WebCore/Modules/speech/SpeechSynthesisEventInit.h:31,
                 from /app/webkit/Source/WebCore/Modules/speech/SpeechSynthesisEvent.h:31,
                 from /app/webkit/WebKitBuild/GTK/Release/WebCore/DerivedSources/JSSpeechSynthesisEvent.h:27,
                 from /app/webkit/WebKitBuild/GTK/Release/WebCore/DerivedSources/JSSpeechSynthesisErrorEvent.h:26,
                 from /app/webkit/WebKitBuild/GTK/Release/WebCore/DerivedSources/JSSpeechSynthesisErrorEvent.cpp:25:
/app/webkit/Source/WebCore/dom/ContextDestructionObserver.h:40:36: error: inline function ‘WebCore::ScriptExecutionContext* WebCore::ContextDestructionObserver::scriptExecutionContext() const’ used but never defined [-Werror]
```

From that build error, it is unclear to me where the include is missing. Also, it builds and links properly if I drop the `inline` 🤷🏻

Reply:
https://ews-build.webkit.org/#/builders/2/builds/34698